internal/provider/base.go

```diff
@@ -32,3 +34,19 @@ type Provider interface {
 	// PostCreateManifestsHook allows the provider to perform additional tasks required after the cluster manifests are created
 	PostCreateManifestsHook(cluster *common.Cluster, envVars *[]string, workDir string) error
 }
+
+func GetMachineNetworkForUserManagedNetworking(log logrus.FieldLogger, cluster *common.Cluster) *[]installcfg.MachineNetwork {
```

Comment:
why do you need to return a pointer? 
Empty arr is nil as well https://go.dev/play/p/qfLUDdPq2Ry 
This way you replace the calls to the functions with just 
```
cfg.Networking.MachineNetwork = provider.GetMachineNetworkForUserManagedNetworking(p.Log, cluster)
```
No need for a variable and all the if else statements

Reply:
I wanted to leave the previous logic the same but I think your solution will also leave it as it is.


```go
	bootstrapCidr := network.GetPrimaryMachineCidrForUserManagedNetwork(cluster, p.Log)
	if bootstrapCidr != "" {
		...
		cfg.Networking.MachineNetwork = machineNetwork
		cfg.Networking.NetworkType = swag.StringValue(cluster.NetworkType)
	} else {
		cfg.Networking.MachineNetwork = nil
	}


```